go-controller/pkg/libovsdb/libovsdb.go

```diff
@@ -25,7 +26,8 @@ import (
 // newClient creates a new client object given the provided config
 // the stopCh is required to ensure the goroutine for ssl cert
 // update is not leaked
-func newClient(cfg config.OvnAuthConfig, dbModel model.ClientDBModel, stopCh <-chan struct{}) (client.Client, error) {
+func newClient(cfg config.OvnAuthConfig, dbModel model.ClientDBModel, promRegistry prometheus.Registerer,
```

Comment:
How would you feel about passing variable length options directly to `newClient`? That way is is more generic. I have a draft where I am already doing so so it might make sense:

`func newClient(cfg config.OvnAuthConfig, dbModel model.ClientDBModel, stopCh <-chan struct{}, opts ...client.Option)`

Reply:
Sounds much better. Can you link a PR here and I can follow and rebase once merged.

Reply:
Is this one, but just a draft.
https://github.com/openshift/ovn-kubernetes/pull/1122

Your PR would surely go in first.

Reply:
Cool. Ill make this change anyway.

Reply:
Done